lib/connections/service.go

```diff
@@ -282,16 +271,11 @@ next:
 		// If rate limiting is set, and based on the address we should
 		// limit the connection, then we wrap it in a limiter.
 
-		limit := s.shouldLimit(c.RemoteAddr())
-
 		wr := io.Writer(c)
-		if limit && s.writeRateLimit != nil {
-			wr = NewWriteLimiter(c, s.writeRateLimit)
-		}
-
 		rd := io.Reader(c)
-		if limit && s.readRateLimit != nil {
-			rd = NewReadLimiter(c, s.readRateLimit)
+		if s.shouldLimit(c.RemoteAddr()) {
+			wr = s.limiter.newWriteLimiter(c)
+			rd = s.limiter.newReadLimiter(c)
 		}
```

Comment:
I say we should wrap all connections and then limit them based on LimitBwInLan which would allow live toggling of that without reconnects?